torchgeo/transforms/color.py

```diff
@@ -30,7 +30,7 @@ class RandomGrayscale(IntensityAugmentationBase2D):
 
     def __init__(
         self,
-        weights: Tensor,
+        weights: Optional[Tensor],
```

Comment:
As a user, I want to use the averaged grayscale by default. I also don't want to have to know how many channels to compute the average. This is different if I actually want to use weights.

Reply:
This is a lot of extra code just for that...

Reply:
It's only a few lines so I don't have to define the weights. I initially made it a separate class. Maybe that's cleaner.

Reply:
Nah, separate class is all code duplication

Reply:
You'll prob want to make this a separate PR to save time. It affects both MoCo and SimCLR

Reply:
Sounds good. I actually thought of a better way to handle this so I'll push an update shortly.

Reply:
I actually didn't see you already it working if `grayscale_weights=None`, in that case, let's just forget this entirely